src/common/components/SupportSidebar/MentoringConsumables.jsx

```diff
@@ -153,18 +154,18 @@ function MentoringConsumables({
         mentorship_service: service?.slug,
       },
     });
-    const relatedConsumables = consumables.find((consumable) => consumable?.mentorship_services?.some((c) => c?.slug === service?.slug));
-    setProgramMentors(data);
+    const relatedConsumable = consumables.find((consumable) => consumable?.mentorship_services?.some((c) => c?.slug === service?.slug));
+    setProgramMentors(mentors);
     setConsumableOfService({
-      ...relatedConsumables,
+      ...relatedConsumable,
       balance: {
-        unit: service?.academy?.available_as_saas === false ? -1 : relatedConsumables?.balance?.unit,
+        unit: (service?.academy?.available_as_saas === false || isAvailableForConsumables === false) ? -1 : relatedConsumable?.balance?.unit,
```

Comment:
When coming from the "choose-program" page, services are assigned the available_as_saas key, as seen here:
```
const getServices = async (userRoles) => {
    if (userRoles?.length > 0) {
      delete axios.defaults.headers.common.Academy;
      const mentorshipPromises = await userRoles.map((role) => bc.mentorship({ academy: role?.academy?.id }, true).getService()
        .then((resp) => {
          const data = resp?.data;
          if (data !== undefined && data.length > 0) {
            return data.map((service) => ({
              ...service,
              academy: {
                id: role?.academy.id,
                available_as_saas: role?.academy?.available_as_saas,
              },
            }));
          }
          return [];
        }));
      const mentorshipResults = await Promise.all(mentorshipPromises);
      const recopilatedServices = mentorshipResults.flat();

      setMentorshipServices({
        isLoading: false,
        data: recopilatedServices,
      });
    }
  };
```

I think this is necessary because a user could have different services from different cohorts, so we need to differentiate between them. However, when we're already in a specific program, there’s no need to run this function again, as the cohort itself already tells us whether available_as_saas is true or false. That’s why I kept the `useEffect `on line 132 in the `mentoring` component. however I understand that is not completelly clear so i will change the name `cohortSessionIsSaaS` for better clarity.